synapse/api/auth.py

```diff
@@ -587,7 +587,10 @@ def _get_user_by_access_token(self, token):
     def _get_user_from_macaroon(self, macaroon_str):
         try:
             macaroon = pymacaroons.Macaroon.deserialize(macaroon_str)
-            self._validate_macaroon(macaroon)
+            self.validate_macaroon(
+                macaroon, "access",
+                [lambda c: c == "guest = true", lambda c: c.startswith("time < ")]
```

Comment:
From `#matrix-dev:matrix.org`, basically, it doesn't matter if we always include `guest = true` since the macaroon verifier never fails if we are missing a caveat (due to how macaroons work). So the question is more: do we want to *allow* there to be a `guest = true` caveat.